File: typescript-examples/quick-recipes/api/capture-screenshots.ts

import { BrowserContext, Page } from "playwright";
import { uploadFileToS3 } from "@intuned/browser";

interface Params {
  // No params needed
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  // Navigate to the page to capture
  await page.goto("https://intunedhq.com/");

  // Capture screenshot as bytes
  const screenshotInBytes = await page.screenshot();

  // Upload the screenshot to S3
  // https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/uploadFileToS3
  const uploadedFile = await uploadFileToS3({
    file: screenshotInBytes,
    fileNameOverride: "screenshot.png",
    contentType: "image/png",
  });

  // Provide a signed URL for retrieval
  const signedUrl = await uploadedFile.getSignedUrl();
  console.log(`Check the screenshot at: ${signedUrl}`);
  return uploadedFile;
}

File: python-examples/quick-recipes/api/capture-screenshots.py
from typing import TypedDict

from intuned_browser import upload_file_to_s3
from playwright.async_api import Page


class Params(TypedDict):
    pass


async def automation(page: Page, params: Params | None = None, **_kwargs):
    # Navigate to the page to capture
    await page.goto("https://intunedhq.com/")

    # Capture screenshot as bytes
    screenshot_in_bytes = await page.screenshot()

    # Upload the screenshot to S3
    # https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/upload_file_to_s3

    uploaded_file = await upload_file_to_s3(
        file=screenshot_in_bytes,
        file_name_override="screenshot.png",
        content_type="image/png",
    )

    # Provide a signed URL for retrieval
    signed_url = await uploaded_file.get_signed_url()
    print(f"Check the screenshot at: {signed_url}")
    return uploaded_file
